central/reports/common/searcher.go

```diff
@@ -0,0 +1,90 @@
+package common
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gogo/protobuf/proto"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	apiV2 "github.com/stackrox/rox/generated/api/v2"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/utils"
+)
+
+// TransformReportStateSearchValues transforms match field queries with Report State == SUCCESS or Report State == "SUCCESS" (Exact match)
+// to a disjunction query Report State == "GENERATED" OR Report State == "DELIVERED" query.
+func TransformReportStateSearchValues(searcher search.Searcher) search.Searcher {
+	return search.FuncSearcher{
+		SearchFunc: func(ctx context.Context, q *v1.Query) ([]search.Result, error) {
+			// Local copy to avoid changing input.
+			local := q.Clone()
+			pagination := local.GetPagination()
+			local.Pagination = nil
+
+			replaceSearchBySuccess(local)
+
+			local.Pagination = pagination
+			return searcher.Search(ctx, local)
+		},
+		CountFunc: func(ctx context.Context, q *v1.Query) (int, error) {
+			// Local copy to avoid changing input.
+			local := q.Clone()
+			pagination := local.GetPagination()
+			local.Pagination = nil
+
+			replaceSearchBySuccess(local)
+
+			local.Pagination = pagination
+			return searcher.Count(ctx, local)
+		},
+	}
+}
+
+// Replace match field queries of type Report State == SUCCESS and Report State == "SUCCESS" (Exact match)
+// Note : This func does no replacement
+//  1. if the query is a prefix query designed to match any non-complete prefix of 'SUCCESS'
+//  2. if the query is a regex query designed to match 'SUCCESS' among other matches
+//  3. if the query is a negation query designed to not match 'SUCCESS' or any of its prefixes
+//
+// To avoid these discrepancies we should stop supporting 'SUCCESS' as a value for "Report State" field in the long run.
```

Comment:
@charmik-redhat, please add a backend and frontend task to stop the use of SUCCESS in the API and UI.   We did not want to do it tech preview due to time constraints but we would like us to tackle this problem in GA.
BE: https://issues.redhat.com/browse/ROX-18979
FE: https://issues.redhat.com/browse/ROX-18980